runtime/src/main/java/org/corfudb/runtime/object/transactions/AbstractTransactionalContext.java

```diff
@@ -252,14 +253,46 @@ public void abortTransaction(TransactionAbortedException ae) {
     }
 
     /**
-     * Retrieves the current timestamp from the sequencer.
+     * Retrieves the current tail from the sequencer, if this
+     * is a nested transaction, then inherit the snapshot
+     * time from the parent transaction.
+     *
      * @return the current global tail
      */
-    public long obtainSnapshotTimestamp() {
-        long timestamp = builder.runtime
-                .getSequencerView().query().getToken().getTokenValue();
-        log.trace("obtainSnapshotTimestamp: SnapshotTimestamp[{}] {}", this, timestamp);
-        return timestamp;
+    private long obtainSnapshotTimestamp() {
+        //TODO(Maithem): Should we check if a transactons with diffent ids are
```

Comment:
typos (transactions and different)...

I'm not sure I understand this TODO, don't all transactions even nested transactions  have their own id's? a new is created on TXBegin...

Reply:
I wanted to make sure that nested transactions inherit their transaction id too, since a nested transaction is really the same transaction. I'll leave it for now.